### PR TITLE
chore: Add Azure Writer example and docs

### PR DESCRIPTION
**Pull request by mdrakiburrahman**
# Why this change is needed

Closes issue: https://github.com/delta-incubator/delta-dotnet/issues/76

# How

- Updates the local project to use [`TokenCredential`](https://learn.microsoft.com/en-us/dotnet/api/azure.core.tokencredential?view=azure-dotnet), which works for many Entra Identity sources, including Visual Studio, Azure CLI, Managed Identity etc.
- Adds a short README and screenshots
- Updates the demo project to write 2 columns
- Add docs on how to connect to other clouds, AWS/GCP etc

# Test

- See README, ran locally and to ADLS

### CODE REVIEW

**Review by MrPowers — DISMISSED** [review]
LGTM

**Comment by mdrakiburrahman** [comment]
Thanks @MrPowers!

Looks like a maintainer needs to approve the CICD pipelines to run:

image

**Comment by mightyshazam** [comment]
Outside of the VisualStudio auth, this should work because it is one of the examples provided in [kafka-delta-ingest](https://github.com/delta-io/kafka-delta-ingest) if you scroll to the `Azure` section. It is pretty environment variable dependent, though.
Object store supports all of the other authentication methods, and the list of environment variables can be found [here](https://github.com/apache/arrow-rs/blob/f0e39cc3cb2f61bbd8294119f4ad9f39545b99f4/object_store/src/azure/builder.rs#L198)

**Comment by mdrakiburrahman** [comment]
@mightyshazam - gotcha, I'll update the README with the links you shared.

Unfortunately it doesn't look like the env vars support setting JWTs, which is probably the most secured option - which is why I included that in the example. We can also use `DefaultAzureCredential` which is a superset of `VisualStudioCredential`.

**Comment by mdrakiburrahman** [comment]
@mightyshazam:

Made 2 changes:

1. Only use `VisualStudioCredentials` if running in Visual Studio, else use [`DefaultAzureCredential`](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet) - which supports a large number of other auth sources
2. Added env-vars `object-stores` supports in the README

**Review by mightyshazam — APPROVED** [review]

File: examples/local/Program.cs [hunk 1]
```diff
@@ -43,7 +74,8 @@ public static async Task Main(string[] args)
                     {
                         ["delta.dataSkippingNumIndexedCols"] = "32",
                         ["delta.setTransactionRetentionDuration"] = null,
```
[the remark this answers is not in the document]
Comment (mdrakiburrahman) [reply examples/local/Program.cs]: Ah I did in the latest push (it was failing a test). I wonder if your browser might be cached haha

[the remark this answers is not in the document]
Comment (mdrakiburrahman) [reply examples/local/Program.cs]: Had removed the null and updated the dictionary to `string` here:

https://github.com/delta-incubator/delta-dotnet/pull/81/commits/fcf8929124759cb4463c6a8e19043fd782504e90

**Pull request merged**